javaparser-core/src/main/java/com/github/javaparser/printer/PrintableSource.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2011, 2013-2020 The JavaParser Team.
+ *
+ * This file is part of JavaParser.
+ *
+ * JavaParser can be used either under the terms of
+ * a) the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ * b) the terms of the Apache License
+ *
+ * You should have received a copy of both licenses in LICENCE.LGPL and
+ * LICENCE.APACHE. Please refer to those files for details.
+ *
+ * JavaParser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ */
+
+package com.github.javaparser.printer;
+
+import com.github.javaparser.Position;
+
+public interface PrintableSource {
+
+    /**
+     * Add the default indentation to the current indentation and push it on the indentation stack.
+     * Does not actually output anything.
+     */
+    PrintableSource indent();
+
+    /**
+     * Add to the current indentation until it is reaches "column" and push it on the indentation stack.
+     * Does not actually output anything.
+     */
+    PrintableSource indentWithAlignTo(int column);
+
+    /**
+     * Pop the last indentation of the indentation stack.
+     * Does not actually output anything.
+     */
+    PrintableSource unindent();
+
+    /**
+     * Append the source string passed as argument to the buffer.
+     * If this is being appended at the beginning of a line, performs indentation first.
+     * <p>
+     * The source line to be printed should not contain newline/carriage-return characters;
+     * use {@link #println(String)} to automatically append a newline at the end of the source string.
+     * If the source line passed as argument contains newline/carriage-return characters would
+     * impredictably affect a correct computation of the current {@link #getCursor()} position.
+     *
+     * @param arg source line to be printed (should not contain newline/carriage-return characters)
```

Comment:
shall we strengthen "should not" to "must not"?

Reply:
idem